corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js

```diff
@@ -95,11 +95,10 @@ hqDefine("cloudcare/js/formplayer/menus/api", function () {
                                 'showError',
                                 errors.LOCK_TIMEOUT_ERROR
                             );
-                        } else if (response.status === 401) {
+                        } else if (response.status === 401 || response.status === 403) {
                             FormplayerFrontend.trigger(
                                 'showError',
-                                formEntryUtils.reloginErrorHtml(),
-                                true
+                                gettext("Could not authenticate, please log in and try again")
```

Comment:
I'm not sure this message is correct for 403, which means "You are authenticated, but you do not have permission to retrieve this resource."

Reply:
@millerdev While technically true that 401 is Unauthorized, I find that is an older usage (coming directly from the server itself). For example, failing a HTTP Basic Authentication could return a 401.

However, more modern systems (including middleware -- Django/Spring Boot) return a 403 for being unauthorized. This is because they are authorized to connect to the server, but then in the process of getting content it has been determined the user does not have access to the requested content.

And further proof is the reason for this PR: the "Unable to connect" message is showing specifically when a user logs out in another tab and tries to nav in a second tab in a web app. So they are "unauthenticated" (403) from CommCare's perspective, thus making this message relevant.

Reply:
On further investigation, I should have responded differently. The PR description says:

> When a navigate_menu request is sent out while the user is unauthenticated, it will return a 403 Forbidden code.

Just so we are clear, in this case "it" refers to "Formplayer." I think this is a bug in Formplayer, not HQ. Formplayer should be returning a 401 if the client is unauthenticated, not a 403.

From [Mozilla's docs on 403](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403):

> for the **403 Forbidden** status code, _re-authenticating makes no difference_.

In this case 401 sounds more appropriate since, according to [the docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401)

> in situations resulting in this status code, _user authentication can allow access to the resource_.